datafusion/src/scalar.rs

```diff
@@ -17,28 +17,29 @@
 
 //! This module provides ScalarValue, an enum that can be used for storage of single elements
 
-use std::{convert::TryFrom, fmt, iter::repeat, sync::Arc};
-
-use arrow::datatypes::{ArrowDictionaryKeyType, DataType, Field, IntervalUnit, TimeUnit};
-use arrow::{
-    array::*,
-    datatypes::{
-        ArrowNativeType, Float32Type, Int16Type, Int32Type, Int64Type, Int8Type,
-        TimestampNanosecondType, UInt16Type, UInt32Type, UInt64Type, UInt8Type,
-    },
-};
+use crate::error::{DataFusionError, Result};
 use arrow::{
     array::{
-        ArrayRef, Int16Builder, Int32Builder, Int64Builder, Int8Builder, ListBuilder,
-        TimestampMicrosecondArray, TimestampMillisecondArray, TimestampNanosecondArray,
-        UInt16Builder, UInt32Builder, UInt64Builder, UInt8Builder,
+        new_null_array, Array, ArrayRef, BinaryArray, BooleanArray, BooleanBuilder,
```

Comment:
maybe at this point it could be more effective to use `use arrow::array::*` and `use arrow::datatypes::*`?

Reply:
if it's okay.

i'll revert back to it.

Reply:
My understanding is that in Rust this idiom is not a concern (as opposed to Python), because all of these will be only on the module.

In Python `import * from X` would be the equivalent of `pub use arrow::array::*`, which I would agree is an anti-pattern due to the public surface.